25_exceptions_lab/02.py

```diff
@@ -1,7 +1,14 @@
-"""
-Write a program that takes a file name
-and prints line count for the file
-
-Alert the user politely if there was any problem opening the file
-"""
-
+"""
+Elad Agam - Exceptions 
+02 - open file and print number of lines - assert if not exists 
+"""
+import os
+import sys 
+
+
+if not os.path.isfile(sys.argv[1]):
+    raise ValueError("%s does not exist\n" % fin)
+
+with open(sys.argv[1],"r") as fin:
+    print sum(1 for _ in fin)
+

```

Comment:
I think "politely" here means without an exception, but also what happens if the file exists but you don't have rights to read it?

I'd not use the 'sum' function here, instead I'd just increment a variable for every line because I think the above code will generate a list of as many 1s as there are lines which means it might be a really long list.